### ISSUE TITLE
Acceptance testing provider using OpenTofu CLI

### ISSUE THREAD

**Issue by janosdebugs** [opentofu/opentofu#977]
### Discussed in https://github.com/orgs/opentofu/discussions/975

<div type='discussions-op-text'>

<sup>Originally posted by **alexhung** December  6, 2023</sup>
Hi, I maintain a few providers for my employer and we are investing how to support OpenTofu in our development process. One main issue we come across is executing acceptance tests using OpenTofu CLI.

I set the env var `TF_ACC_TERRAFORM_PATH` per [documentation](https://github.com/orgs/opentofu/discussions/975) with the path to the OpenTofu CLI (`/usr/local/bin/tofu`). When I run the acceptance tests with no change to our code, I got the following error:

```
=== RUN   TestAccRemoteUpgradeFromVersionWithNoDisableProxyAttr
    resource_artifactory_remote_repository_test.go:42: TestStep 2/2 running init: exit status 1
        Error parsing %!q(PANIC=String method: called String on zero-value addrs.Provider) as a provider address: Invalid provider namespace: The legacy provider namespace "-" can be used only with hostname registry.opentofu.org.
    testing_new.go:79: Error retrieving state, there may be dangling resources: exit status 1
        Error parsing %!q(PANIC=String method: called String on zero-value addrs.Provider) as a provider address: Invalid provider namespace: The legacy provider namespace "-" can be used only with hostname registry.opentofu.org.
```

Is there extra steps I need to do or code to change in order to test against OpenTofu CLI?

Would there be a migration doc/guide for provider developers like for regular users?</div>

**TL;DR:** it seems that the legacy provider namespace (`-`) is used when driving acceptance tests for providers.

**Comment by Yantrio** [comment]
I've responded with a proposed approach to the end user here:
https://github.com/orgs/opentofu/discussions/975#discussioncomment-7777319

Will close if this works for them.

**Comment by janosdebugs** [comment]
Fix works, closing issue and raising a [docs issue](https://github.com/opentofu/opentofu/issues/982).
